radix-engine/src/kernel/call_frame.rs

```diff
@@ -96,17 +96,17 @@ pub struct CallFrame {
     /// Node refs which are immortal during the life time of this frame:
     /// - Any node refs received from other frames;
     /// - Global node refs obtained through substate locking.
-    immortal_node_refs: HashMap<RENodeId, RENodeRefData>,
+    immortal_node_refs: NonIterMap<RENodeId, RENodeRefData>,
 
     /// Node refs obtained through substate locking, which will be dropped upon unlocking.
-    temp_node_refs: HashMap<RENodeId, u32>,
+    temp_node_refs: NonIterMap<RENodeId, u32>,
 
     /// Owned nodes which by definition must live on heap
     /// Also keeps track of number of locks on this node
-    owned_root_nodes: HashMap<RENodeId, u32>,
+    owned_root_nodes: IndexMap<RENodeId, u32>,
 
     next_lock_handle: LockHandle,
-    locks: HashMap<LockHandle, SubstateLock>,
+    locks: IndexMap<LockHandle, SubstateLock>,
```

Comment:
We can use `NoIterMap` for `owned_root_nodes` and `locks`?

Reply:
We iterate over `owned_root_nodes` at:
https://github.com/radixdlt/radixdlt-scrypto/blob/fe73820af3c6709aba58ed75ac371b09fd9c4bf7/radix-engine/src/kernel/call_frame.rs#L550-L551

We iterate over `locks` at:
https://github.com/radixdlt/radixdlt-scrypto/blob/fe73820af3c6709aba58ed75ac371b09fd9c4bf7/radix-engine/src/kernel/call_frame.rs#L445-L447

(I was not analyzing whether the particular iteration has potential of causing determinism or not; I simply followed the migration algo described at https://github.com/radixdlt/radixdlt-scrypto/pull/931#issue-1644373644)